test/unit/foreman_ansible/fact_parser_test.rb

```diff
@@ -0,0 +1,282 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+module ForemanAnsible
+  # Checks sample Ansible facts to see if it can assign them to
+  # Host properties
+  class FactParserTest < ActiveSupport::TestCase
+    setup do
+      facts_json = HashWithIndifferentAccess.new(read_json_fixture('facts/ansible_facts.json'))
+      @facts_parser = AnsibleFactParser.new(facts_json)
+    end
+
+    test 'finds facter domain even if ansible_domain is empty' do
+      expect_where(Domain, @facts_parser.facts[:facter_domain])
+      @facts_parser.domain
+    end
+
+    test 'finds model' do
+      expect_where(Model, @facts_parser.facts[:ansible_product_name])
+      @facts_parser.model
+    end
+
+    test 'finds architecture' do
+      expect_where(Architecture, @facts_parser.facts[:ansible_architecture])
+      @facts_parser.architecture
+    end
+
+    test 'does not set environment' do
+      refute @facts_parser.environment
+    end
+
+    test 'calculates virtual reported data' do
+      refute @facts_parser.virtual
+    end
+
+    test 'calculates ram reported data' do
+      assert_equal 7899, @facts_parser.ram
+    end
+
+    test 'calculates sockets reported data' do
+      assert_equal 1, @facts_parser.sockets
+    end
+
+    test 'calculates cores reported data' do
+      assert_equal 2, @facts_parser.cores
+    end
+
+    test 'creates operatingsystem from operating system options' do
+      sample_mock = mock
+      major_fact = @facts_parser.facts['ansible_distribution_major_version']
+      _, minor_fact = @facts_parser.
+                      facts['ansible_distribution_version'].split('.')
+      Operatingsystem.expects(:where).
+        with(:name => @facts_parser.facts['ansible_distribution'],
+             :major => major_fact, :minor => minor_fact || '').
+        returns(sample_mock)
+      sample_mock.expects(:first)
+      @facts_parser.operatingsystem
+    end
+
+    test 'does not fail if facts are not enough to create OS' do
+      @facts_parser.expects(:os_name).returns('fakeos').at_least_once
+      @facts_parser.expects(:os_major).returns('').at_least_once
+      @facts_parser.expects(:os_minor).returns('').at_least_once
+      @facts_parser.expects(:os_description).returns('').at_least_once
+      Operatingsystem.any_instance.expects(:valid?).returns(false)
+      assert_nil @facts_parser.operatingsystem
+    end
+
+    private
+
+    def expect_where(model, fact_name)
+      sample_mock = mock
+      model.expects(:where).
+        with(:name => fact_name).
+        returns(sample_mock)
+      sample_mock.expects(:first_or_create)
+    end
+  end
+
+  # Tests for Network parser
+  class NetworkFactParserTest < ActiveSupport::TestCase
+    setup do
+      @facts_parser = AnsibleFactParser.new(
+        HashWithIndifferentAccess.new(
+          '_type' => 'ansible',
```

Comment:
Ansible? is this a left over? I suggest to move these fixtures into separate files, we will be adding more over time.

Reply:
Well, PR is based on commit of adding Ansible parser due to using own initializer defined in Ansible PR.